packages/@aws-cdk/core/lib/construct-compat.ts

```diff
@@ -522,7 +522,6 @@ export class ConstructNode {
    * Remove the child with the given name, if present.
    *
    * @returns Whether a child with the given name was deleted.
-   * @experimental
```

Comment:
Since some files do not exist in v2-main (like this one), this will result in some merge conflicts. @NetaNir can you please make sure to do a forward merge to v2-main immediately after this is merged so that conflicts will be limited to this change and not intermingled with lots of future changes? (copy @nija-at)